uvls/src/main.rs

```diff
@@ -389,6 +392,27 @@ impl LanguageServer for Backend {
                 self.pipeline.touch(&uri);
                 self.client.code_lens_refresh().await?;
             }
+            "uvls/generate_diagram" => {
+                let root_fileid = FileID::from_uri(&Url::parse(uri.as_str()).unwrap());
+                let root_graph = self.pipeline.root().borrow_and_update().clone();
+                if !root_graph.contains_id(root_fileid){{}}
+                
+                let document = root_graph.files.get(&root_fileid).unwrap();
+                let g = ast::graph::Graph::new(document.source.clone(), document.tree.clone(), uri.clone());
+
+                // write graph script (dot) to file:
+                let diagram_file_extension = "dot";
+                let re = regex::Regex::new(r"(.*\.)(.*)").unwrap();
+                let path = re.replace(uri.path(), |caps: &regex::Captures| {format!("{}{}", &caps[1], diagram_file_extension)});
+                let file = std::fs::File::create(path.as_ref())
```

Comment:
We might also think about not creating an additional file at all.  
Or maybe adding a setting for this.  
Creating a file might be useful for editing the graph later on.

Reply:
How would this work without creating a .dot file? Or what do you mean?

Reply:
We can simply return the Graph-Script (as we already do as soon as the file creation fails) and pass it via our vsCode extension to the GraphViz Extension that is able to display it without having to read a `.dot` file.  
However, this prevents the end-user to edit the Graph-Script in any custom way. That's why I think this should only be a fallback solution.